src/main/java/com/epam/edumanagementsystem/parent/rest/api/ParentController.java

```diff
@@ -106,7 +107,15 @@ public String editParent(@ModelAttribute("parent") @Valid ParentEditDto parentDt
     @PostMapping("/{id}/image/add")
     @Operation(summary = "Adds image to selected parent's profile")
     public String addImage(@PathVariable("id") Long id, @RequestParam("picture") MultipartFile multipartFile) {
-        parentService.addImage(parentService.findById(id), multipartFile);
+        if (parentService.existsParentById(id)) {
+            ParentDto parent = parentService.findById(id);
+            if (parent.getPicUrl() != null) {
+                imageService.deleteImage(parent.getPicUrl());
+                parentService.addImage(parent, multipartFile);
```

Comment:
Remove parentService.addImage from if/else blocks as it is executed in both clauses, no else block is needed